internal/agentcfg/fetch.go

```diff
@@ -17,38 +17,46 @@
 
 package agentcfg
 
-import (
-	"context"
-
-	"github.com/elastic/apm-server/internal/beater/config"
-)
-
-// TransactionSamplingRateKey is the agent configuration key for the
-// sampling rate. This is used by the Jaeger handler to adapt our agent
-// configuration to the Jaeger remote sampler protocol.
-const TransactionSamplingRateKey = "transaction_sample_rate"
+import "context"
 
 // Fetcher defines a common interface to retrieving agent config.
 type Fetcher interface {
 	Fetch(context.Context, Query) (Result, error)
 }
 
-// DirectFetcher is an agent config fetcher which serves requests out of a
-// statically defined set of agent configuration. These configurations are
-// typically provided via Fleet.
-type DirectFetcher struct {
-	cfgs []AgentConfig
+// AgentConfig holds an agent configuration definition.
+type AgentConfig struct {
+	// ServiceName holds the service name to which this agent configuration
+	// applies. This is optional.
+	ServiceName string
+
+	// ServiceEnvironment holds the service environment to which this agent
+	// configuration applies. This is optional.
+	ServiceEnvironment string
+
+	// AgentName holds the agent name to which this agent configuration
+	// applies. This is optional, and is used for filtering configuration
+	// settings for unauthenticated agents.
+	AgentName string
+
+	// Etag holds a unique ID for the configuration, which agents
+	// will send along with their queries. The server uses this to
+	// determine whether agent configuration has been applied.
+	Etag string
+
+	// Config holds configuration settings that should be sent to
+	// agents matching the above constraints.
+	Config map[string]string
 }
 
-// NewDirectFetcher returns a new DirectFetcher that serves agent configuration
-// requests using cfgs.
-func NewDirectFetcher(cfgs []AgentConfig) *DirectFetcher {
-	return &DirectFetcher{cfgs}
+func NewEmptyFetcher() Fetcher {
+	return &emptyFetcher{}
```

Comment:
Exposing this Fetcher here although it should only be used for testing is not ideal; IMO there should at least be a comment. 
Alternatively you could export `zeroResult()` and then add the emptyFetcher directly in the test functions where it is needed; also not great though.

Reply:
Ah right, ignore my comment then.